rest/server_context.go

```diff
@@ -655,7 +655,9 @@ func dbcOptionsFromConfig(ctx context.Context, sc *ServerContext, config *DbConf
 	}
 	importOptions.BackupOldRev = base.BoolDefault(config.ImportBackupOldRev, false)
 
-	if config.ImportPartitions == nil {
+	if config.ImportPartitions == nil && sc.Config.IsServerless() {
```

Comment:
It's a bit of a style nit, but I think I would implement a `DefaultImportPartitions(serverless bool) uint16` function in base to encapsulate the handling here and above, particularly given the complexity it's adding to this particular if block.